var/spack/repos/builtin/packages/bison/package.py

```diff
@@ -35,7 +35,8 @@ class Bison(AutotoolsPackage):
 
     version('3.0.4', 'a586e11cd4aff49c3ff6d3b6a4c9ccf8')
 
-    depends_on('m4', type='build')
+    # This is a run-time dependency as well
+    depends_on('m4')
```

Comment:
Default is `build, link`. See [documentation](http://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-specs)